Could not resolve lightningcss.linux-arm64-gnu.node in Docker with M1

I'm not sure if this belongs in Parcel or here, but I'll start here and you can correct me.

It seems like trying to use `lightningcss` inside Docker on an M1 Mac fails to do the right thing with the binary dependencies (not sure if M1 is related or not, but that's what I'm using).  I've made a [demo repo to show the problem](https://github.com/humphd/parcel-m1-docker-bug).

If I do `npm run build` natively on my M1 Mac, it works.  However, inside Docker, the build gives this:

```sh
❯ docker build . -t parcel-bug
[+] Building 21.3s (9/9) FINISHED
 => [internal] load build definition from Dockerfile                       0.0s
 => => transferring dockerfile: 116B                                       0.0s
 => [internal] load .dockerignore                                          0.0s
 => => transferring context: 34B                                           0.0s
 => [internal] load metadata for docker.io/library/node:16.18.1-bullseye   0.3s
 => [1/5] FROM docker.io/library/node:16.18.1-bullseye@sha256:7fb41fa8410  0.0s
 => [internal] load build context                                          0.0s
 => => transferring context: 329B                                          0.0s
 => CACHED [2/5] WORKDIR /app                                              0.0s
 => [3/5] COPY . .                                                         0.0s
 => [4/5] RUN npm i                                                       19.6s
 => ERROR [5/5] RUN npm run build                                          1.2s
------
 > [5/5] RUN npm run build:
#0 0.389
#0 0.389 > parcel-bug@1.0.0 build
#0 0.389 > parcel build index.html
#0 0.389
#0 0.819 Building...
#0 1.102 🚨 Build failed.
#0 1.104
#0 1.105 @parcel/transformer-css: Could not resolve module
#0 1.105 "/app/node_modules/lightningcss/lightningcss.linux-arm64-gnu.node" from
#0 1.105 "/app/node_modules/lightningcss/node/index.js"
#0 1.106
#0 1.106   Error: Could not resolve module
#0 1.106   "/app/node_modules/lightningcss/lightningcss.linux-arm64-gnu.node" from
#0 1.106   "/app/node_modules/lightningcss/node/index.js"
#0 1.106       at $0578d0f6e116167e$export$fb2a0b866a8162dc.resolve
#0 1.106   (/app/node_modules/@parcel/package-manager/lib/index.js:4991:21)
#0 1.106       at NodePackageManager.resolveSync
#0 1.106   (/app/node_modules/@parcel/package-manager/lib/index.js:3510:42)
#0 1.106       at NodePackageManager.requireSync
#0 1.106   (/app/node_modules/@parcel/package-manager/lib/index.js:3352:34)
#0 1.106       at Module.m.require
#0 1.106   (/app/node_modules/@parcel/package-manager/lib/index.js:3366:25)
#0 1.106       at require (node:internal/modules/cjs/helpers:103:18)
#0 1.106       at Object.<anonymous> (/app/node_modules/lightningcss/node/index.js:21:22)
#0 1.106       at Module._compile (node:internal/modules/cjs/loader:1155:14)
#0 1.106       at Object.Module._extensions..js
#0 1.106   (node:internal/modules/cjs/loader:1209:10)
#0 1.106       at Module.load (node:internal/modules/cjs/loader:1033:32)
#0 1.106       at NodePackageManager.load
#0 1.106   (/app/node_modules/@parcel/package-manager/lib/index.js:3375:15)
#0 1.108
------
ERROR: failed to solve: executor failed running [/bin/sh -c npm run build]: exit code: 1
```

Inside the container, I can see that it's pulled in what I think are the right dependencies:

```sh
/app/node_modules# ls lightningcss*
lightningcss:
LICENSE  README.md  node  package.json

lightningcss-linux-arm64-gnu:
README.md  lightningcss.linux-arm64-gnu.node  package.json

lightningcss-linux-arm64-musl:
README.md  lightningcss.linux-arm64-musl.node  package.json
```

Inside the container, node is using `process.platform === 'linux'` and `process.arch === 'arm64'`.

This issue occurs without anything fancy, just `npm` on Node.js v18.12.0, and it occurs on linux as well:

<details>
<summary>Console output (Node.js v18.12.0)</summary>

```
  const err = new Error(message);
              ^

Error: Cannot find module '../lightningcss.linux-x64-gnu.node'
Require stack:
- /…/node_modules/lightningcss/node/index.js
- /…/node_modules/esbuild-css-modules-plugin/lib/plugin.js
- /…/node_modules/esbuild-css-modules-plugin/index.js
    at Module._resolveFilename (node:internal/modules/cjs/loader:995:15)
    at Module._load (node:internal/modules/cjs/loader:841:27)
    at Module.require (node:internal/modules/cjs/loader:1061:19)
    at require (node:internal/modules/cjs/helpers:103:18)
    at Object.<anonymous> (/…/node_modules/lightningcss/node/index.js:21:22)
    at Module._compile (node:internal/modules/cjs/loader:1159:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1213:10)
    at Module.load (node:internal/modules/cjs/loader:1037:32)
    at Module._load (node:internal/modules/cjs/loader:878:12)
    at Module.require (node:internal/modules/cjs/loader:1061:19) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/…/node_modules/lightningcss/node/index.js',
    '/…/node_modules/esbuild-css-modules-plugin/lib/plugin.js',
    '/…/node_modules/esbuild-css-modules-plugin/index.js'
  ]
}
```
</details>

![image](https://user-images.githubusercontent.com/101106010/202747303-9bd552db-9e65-4f39-9573-51307465173f.png)

--- a/.github/workflows/release.yml
+++ b/.github/workflows/release.yml
@@ -1,7 +1,6 @@
 name: release
 on:
   workflow_dispatch:
-  push:
 
 jobs:
   build:
@@ -134,8 +133,6 @@ jobs:
 
     steps:
       - uses: actions/checkout@v2
-      - name: Debug
-        run: ldd --version
       - name: Install Node.JS
         uses: actions/setup-node@v2
         with:
@@ -174,76 +171,76 @@ jobs:
             *.node
             lightningcss
 
-  # build-wasm:
-  #   runs-on: ubuntu-latest
-  #   steps:
-  #     - uses: actions/checkout@v2
-  #     - name: Install Node.JS
-  #       uses: actions/setup-node@v2
-  #       with:
-  #         node-version: 14
-  #     - name: Install wasm-pack
-  #       run: curl https://rustwasm.github.io/wasm-pack/installer/init.sh -sSf | sh
-  #     - name: Build wasm
-  #       run: yarn wasm-browser:build-release
-  #     - name: Upload artifacts
-  #       uses: actions/upload-artifact@v2
-  #       with:
-  #         name: wasm
-  #         path: node/pkg
+  build-wasm:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install Node.JS
+        uses: actions/setup-node@v2
+        with:
+          node-version: 14
+      - name: Install wasm-pack
+        run: curl https://rustwasm.github.io/wasm-pack/installer/init.sh -sSf | sh
+      - name: Build wasm
+        run: yarn wasm-browser:build-release
+      - name: Upload artifacts
+        uses: actions/upload-artifact@v2
+        with:
+          name: wasm
+          path: node/pkg
 
-  # release:
-  #   runs-on: ubuntu-latest
-  #   name: Build and release
-  #   needs:
-  #     - build
-  #     - build-linux
-  #     - build-apple-silicon
-  #     - build-wasm
-  #   steps:
-  #     - uses: actions/checkout@v1
-  #     - uses: bahmutov/npm-install@v1.1.0
-  #     - name: Download artifacts
-  #       uses: actions/download-artifact@v2
-  #       with:
-  #         path: artifacts
-  #     - name: Build npm packages
-  #       run: |
-  #         node scripts/build-npm.js
-  #         node scripts/build-wasm.js
-  #     - run: echo //registry.npmjs.org/:_authToken=${NPM_TOKEN} > ~/.npmrc
-  #       env:
-  #         NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
-  #     - name: Publish to npm
-  #       run: |
-  #         for pkg in npm/*; do
-  #           echo "Publishing $pkg..."
-  #           cd $pkg;
-  #           npm publish;
-  #           cd ../..;
-  #         done
-  #         cd cli
-  #         echo "Publishing lightningcss-cli...";
-  #         npm publish
-  #         cd ..
-  #         echo "Publishing lightningcss...";
-  #         npm publish
+  release:
+    runs-on: ubuntu-latest
+    name: Build and release
+    needs:
+      - build
+      - build-linux
+      - build-apple-silicon
+      - build-wasm
+    steps:
+      - uses: actions/checkout@v1
+      - uses: bahmutov/npm-install@v1.1.0
+      - name: Download artifacts
+        uses: actions/download-artifact@v2
+        with:
+          path: artifacts
+      - name: Build npm packages
+        run: |
+          node scripts/build-npm.js
+          node scripts/build-wasm.js
+      - run: echo //registry.npmjs.org/:_authToken=${NPM_TOKEN} > ~/.npmrc
+        env:
+          NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
+      - name: Publish to npm
+        run: |
+          for pkg in npm/*; do
+            echo "Publishing $pkg..."
+            cd $pkg;
+            npm publish;
+            cd ../..;
+          done
+          cd cli
+          echo "Publishing lightningcss-cli...";
+          npm publish
+          cd ..
+          echo "Publishing lightningcss...";
+          npm publish
 
-  # release-crates:
-  #   runs-on: ubuntu-latest
-  #   name: Release Rust crate
-  #   steps:
-  #     - uses: actions/checkout@v1
-  #     - uses: bahmutov/npm-install@v1.1.0
-  #     - name: Install Rust
-  #       uses: actions-rs/toolchain@v1
-  #       with:
-  #         toolchain: stable
-  #         profile: minimal
-  #         override: true
-  #     - run: cargo login ${CRATES_IO_TOKEN}
-  #       env:
-  #         CRATES_IO_TOKEN: ${{ secrets.CRATES_IO_TOKEN }}
-  #     - run: |
-  #         cargo install cargo-workspaces
-  #         cargo workspaces publish --from-git -y
+  release-crates:
+    runs-on: ubuntu-latest
+    name: Release Rust crate
+    steps:
+      - uses: actions/checkout@v1
+      - uses: bahmutov/npm-install@v1.1.0
+      - name: Install Rust
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
+          profile: minimal
+          override: true
+      - run: cargo login ${CRATES_IO_TOKEN}
+        env:
+          CRATES_IO_TOKEN: ${{ secrets.CRATES_IO_TOKEN }}
+      - run: |
+          cargo install cargo-workspaces
+          cargo workspaces publish --from-git -y

--- a/.github/workflows/release.yml
+++ b/.github/workflows/release.yml
@@ -1,6 +1,7 @@
 name: release
 on:
   workflow_dispatch:
+  push:
 
 jobs:
   build:
@@ -148,6 +149,9 @@ jobs:
         if: ${{ matrix.setup }}
         run: ${{ matrix.setup }}
 
+      - name: Debug
+        run: ldd --version
+
       - name: Setup rust target
         run: rustup target add ${{ matrix.target }}
 
@@ -171,76 +175,76 @@ jobs:
             *.node
             lightningcss
 
-  build-wasm:
-    runs-on: ubuntu-latest
-    steps:
-      - uses: actions/checkout@v2
-      - name: Install Node.JS
-        uses: actions/setup-node@v2
-        with:
-          node-version: 14
-      - name: Install wasm-pack
-        run: curl https://rustwasm.github.io/wasm-pack/installer/init.sh -sSf | sh
-      - name: Build wasm
-        run: yarn wasm-browser:build-release
-      - name: Upload artifacts
-        uses: actions/upload-artifact@v2
-        with:
-          name: wasm
-          path: node/pkg
+  # build-wasm:
+  #   runs-on: ubuntu-latest
+  #   steps:
+  #     - uses: actions/checkout@v2
+  #     - name: Install Node.JS
+  #       uses: actions/setup-node@v2
+  #       with:
+  #         node-version: 14
+  #     - name: Install wasm-pack
+  #       run: curl https://rustwasm.github.io/wasm-pack/installer/init.sh -sSf | sh
+  #     - name: Build wasm
+  #       run: yarn wasm-browser:build-release
+  #     - name: Upload artifacts
+  #       uses: actions/upload-artifact@v2
+  #       with:
+  #         name: wasm
+  #         path: node/pkg
 
-  release:
-    runs-on: ubuntu-latest
-    name: Build and release
-    needs:
-      - build
-      - build-linux
-      - build-apple-silicon
-      - build-wasm
-    steps:
-      - uses: actions/checkout@v1
-      - uses: bahmutov/npm-install@v1.1.0
-      - name: Download artifacts
-        uses: actions/download-artifact@v2
-        with:
-          path: artifacts
-      - name: Build npm packages
-        run: |
-          node scripts/build-npm.js
-          node scripts/build-wasm.js
-      - run: echo //registry.npmjs.org/:_authToken=${NPM_TOKEN} > ~/.npmrc
-        env:
-          NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
-      - name: Publish to npm
-        run: |
-          for pkg in npm/*; do
-            echo "Publishing $pkg..."
-            cd $pkg;
-            npm publish;
-            cd ../..;
-          done
-          cd cli
-          echo "Publishing lightningcss-cli...";
-          npm publish
-          cd ..
-          echo "Publishing lightningcss...";
-          npm publish
+  # release:
+  #   runs-on: ubuntu-latest
+  #   name: Build and release
+  #   needs:
+  #     - build
+  #     - build-linux
+  #     - build-apple-silicon
+  #     - build-wasm
+  #   steps:
+  #     - uses: actions/checkout@v1
+  #     - uses: bahmutov/npm-install@v1.1.0
+  #     - name: Download artifacts
+  #       uses: actions/download-artifact@v2
+  #       with:
+  #         path: artifacts
+  #     - name: Build npm packages
+  #       run: |
+  #         node scripts/build-npm.js
+  #         node scripts/build-wasm.js
+  #     - run: echo //registry.npmjs.org/:_authToken=${NPM_TOKEN} > ~/.npmrc
+  #       env:
+  #         NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
+  #     - name: Publish to npm
+  #       run: |
+  #         for pkg in npm/*; do
+  #           echo "Publishing $pkg..."
+  #           cd $pkg;
+  #           npm publish;
+  #           cd ../..;
+  #         done
+  #         cd cli
+  #         echo "Publishing lightningcss-cli...";
+  #         npm publish
+  #         cd ..
+  #         echo "Publishing lightningcss...";
+  #         npm publish
 
-  release-crates:
-    runs-on: ubuntu-latest
-    name: Release Rust crate
-    steps:
-      - uses: actions/checkout@v1
-      - uses: bahmutov/npm-install@v1.1.0
-      - name: Install Rust
-        uses: actions-rs/toolchain@v1
-        with:
-          toolchain: stable
-          profile: minimal
-          override: true
-      - run: cargo login ${CRATES_IO_TOKEN}
-        env:
-          CRATES_IO_TOKEN: ${{ secrets.CRATES_IO_TOKEN }}
-      - run: |
-          cargo install cargo-workspaces
-          cargo workspaces publish --from-git -y
+  # release-crates:
+  #   runs-on: ubuntu-latest
+  #   name: Release Rust crate
+  #   steps:
+  #     - uses: actions/checkout@v1
+  #     - uses: bahmutov/npm-install@v1.1.0
+  #     - name: Install Rust
+  #       uses: actions-rs/toolchain@v1
+  #       with:
+  #         toolchain: stable
+  #         profile: minimal
+  #         override: true
+  #     - run: cargo login ${CRATES_IO_TOKEN}
+  #       env:
+  #         CRATES_IO_TOKEN: ${{ secrets.CRATES_IO_TOKEN }}
+  #     - run: |
+  #         cargo install cargo-workspaces
+  #         cargo workspaces publish --from-git -y

--- a/.github/workflows/release.yml
+++ b/.github/workflows/release.yml
@@ -114,11 +114,11 @@ jobs:
             setup: npm install --global yarn@1
           - target: aarch64-unknown-linux-gnu
             strip: aarch64-linux-gnu-strip
-            image: ghcr.io/napi-rs/napi-rs/nodejs-rust:lts-debian
+            image: ghcr.io/napi-rs/napi-rs/nodejs-rust:lts-debian@sha256:4e170e00b4e33251a775bcadc34380a6a5a86611df132590885257e453646fdc
             setup: apt install gcc-aarch64-linux-gnu g++-aarch64-linux-gnu -y
           - target: armv7-unknown-linux-gnueabihf
             strip: arm-linux-gnueabihf-strip
-            image: ghcr.io/napi-rs/napi-rs/nodejs-rust:lts-debian
+            image: ghcr.io/napi-rs/napi-rs/nodejs-rust:lts-debian@sha256:4e170e00b4e33251a775bcadc34380a6a5a86611df132590885257e453646fdc
             setup: apt install gcc-arm-linux-gnueabihf g++-arm-linux-gnueabihf -y
           - target: aarch64-unknown-linux-musl
             image: ghcr.io/napi-rs/napi-rs/nodejs-rust:lts-alpine
@@ -134,6 +134,8 @@ jobs:
 
     steps:
       - uses: actions/checkout@v2
+      - name: Debug
+        run: ldd --version
       - name: Install Node.JS
         uses: actions/setup-node@v2
         with:
@@ -148,9 +150,6 @@ jobs:
       - name: Setup cross compile toolchain
         if: ${{ matrix.setup }}
         run: ${{ matrix.setup }}
-
-      - name: Debug
-        run: ldd --version
 
       - name: Setup rust target
         run: rustup target add ${{ matrix.target }}

